src/api/auth/AuthApi.ts

```diff
@@ -6,28 +6,58 @@ import {
     type SignUpResponse,
 } from "./types";
 
-const SIGNIN_URL = "auth/signin";
+const SIGN_IN_URL = "auth/signin";
+const SIGN_UP_URL = "auth/signup";
+const SIGN_UP_EMAIL_EXIST_ERROR =
+    "This email had been associated with another account.";
+const SIGN_UP_USERNAME_EXIST_ERROR =
+    "This username had been associated with another account.";
 
 const baseApi = new BaseApi();
 
 const signIn = async (
     usernameOrEmail: string,
     password: string
 ): Promise<SignInResponse> => {
-    return await baseApi.post<SignInRequest, SignInResponse>(SIGNIN_URL, {
+    const response = await baseApi.post<SignInRequest>(SIGN_IN_URL, {
         usernameOrEmail,
         password,
     });
+    return await response.json();
 };
 
 const signUp = async (
-    usernameOrEmail: string,
+    username: string,
+    email: string,
     password: string
-): Promise<SignInResponse> => {
-    return await baseApi.post<SignUpRequest, SignUpResponse>(SIGNIN_URL, {
-        usernameOrEmail,
+): Promise<SignUpResponse> => {
+    const response = await baseApi.post<SignUpRequest>(SIGN_UP_URL, {
+        username,
         password,
+        email,
     });
+    if (response.ok) {
+        return {
+            success: true,
+        };
+    } else {
+        const responseBody: null | {
+            Success: boolean;
+            Body: { Message: string };
+        } = await response.json();
+        console.log(JSON.stringify(responseBody));
+        const message = responseBody?.Body.Message;
+        console.log("a");
+        return {
+            success: false,
+            reason:
```

Comment:
Chỗ này chuyển thành `message` để tiện mapping
Với cả backend đã trả về message hoàn chỉnh rồi thì nên display cho người dùng luôn, nếu message chưa ổn thì có thể liên hệ với BE để update lại